musicbot/bot.py

```diff
@@ -4159,120 +4150,47 @@ async def on_voice_state_update(self, member, before, after):
         else:
             return
 
-        if (
-            member == self.user and not after.channel
-        ):  # if bot was disconnected from channel
-            await self.disconnect_voice_client(before.channel.guild)
+        if member == self.user:
+            if not after.channel:  # if bot was disconnected from channel
+                await self.disconnect_voice_client(before.channel.guild)
```

Comment:
Needless refactor, should leave it as it was.

Reply:
Ah, you're right - I think I needed the return if `member == self.user` in a previous version but that doesn't seem to be the case anymore after factoring out the empty check